File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numMap = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int difference = target - nums[i];
            if(numMap.containsKey(difference)){
                int arr[] = {i, numMap.get(difference)};
                Arrays.sort(arr);
                return arr;
            }else{
                numMap.put(nums[i],i);
            }
        }
        int arr[] ={-1,-1};
        return arr;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l,r = 0, len(nums)-1
        while l < r:
            if(r-l==1):
                return min(nums[r],nums[l])
            mid = (l+r)//2
            if(nums[l] < nums[mid] and nums[r] < nums[mid]):
                l = mid +1
            elif(nums[l] > nums[mid] and nums[r] > nums[mid]):
                r = mid
            elif(nums[l] < nums[mid] < nums[r]):
                return nums[l]
        return nums[l]